nservicebus/scheduling-with-nservicebus.md

```diff
@@ -25,25 +25,20 @@ The difference between these examples is that in the latter a name is given for
+## Current Limitations
 
-When the task is created it is given an unique identifier. The identifier for the task is sent in a message to the Timeout Manager. When it times out and the Timeout Manager returns the message containing the identifier to the endpoint with the scheduled task, the endpoint uses that identifier to fetch and invoke the task from its internal list of tasks.
-
-## Assumptions
-
-- Scheduled tasks are created at endpoint startup using for example a class the implements the `IWantToRunWhenBusStartsAndStops` interface;
-- If the process restarts, all scheduled tasks are recreated and given new identifiers. Tasks scheduled before the restart will not be found and a message is written to the log. This is expected behavior;
-- A scheduled task will run forever, as long as the endpoint is running, there is no way to cancel it except skipping its execution each time is invoked;
+- Since the scheduler is non-durable, if the process restarts, all scheduled tasks (that are created during the endpoint's startup) are recreated and given new identifiers. Tasks scheduled before the restart that arrive at the endpoint queue will not be found and a message is written to the log as information.  
+- Already created schedule tasks cannot be canceled or changed.
+- Does not support a task to be started at a specified specific time.
 - Each scheduled task executes on a new thread using the `Task.Factory.StartNew(Action)` method, which means that there will be no transaction scope by default and it is up to you to create one if needed;
-- You will probably only do a `Bus.Send()` or `Bus.SendLocal()` in the task. The handler of the sent message will be invoked as all other handlers and if configured and supported by underlying transport will be wrapped in a transaction as usual;
+- Since the scheduler uses the queuing mechanism, it does have some side effects on the timeliness of scheduled tasks. When a task is scheduled to be run at a given time it is actual not "executed at that time", it is instead "queued at that time" to be executed. In most cases this distinction will have no noticeable effect on the behavior of the the scheduling API. However in high load systems the fact that a scheduled task is added to the back of the queue can result in a noticeable delay between the "time the task has been request to be run" and the "time the task is actually executed".
+- The scheduler API does not support scaling out the endpoint on the same machine or doing a side-by-side deployment of an endpoint. In both these cases, you have multiple endpoint instances sharing the same queue where the schedule task arrives.  As the scheduler is not-durable and the task details are stored in the endpoint instance that created the task, the task may or may not be found/executed if it is picked up the instance that did not originally create the schedule task. When a task specified by the ID cannot be found, it is no longer re-scheduled to be executed again.

```

Comment:
I understood the above, but I'd say that it is not clear enough, the reader needs to connect the dots understanding that each time a new Task is scheduled a **new** task id is created, thus 2 instances of the same endpoint will create the same task with different ids, then the issue. Correct?